### PR TITLE
General fixes & improvements.

### PR DESCRIPTION
**Pull request by wrefgtzweve**
Changes:

Players now exit spawn protection when they use any kind of entity
Removed halo preview on spawn protected players
Use table vars on players instead of NWBools
Use functions to reduce copy paste code
Remove not needed hook.Remove's
Fixed whitespaces

### CODE REVIEW

File: lua/autorun/server/sv_spawn_protection.lua [hunk 1]
```diff
@@ -188,81 +176,80 @@ local function setSpawnProtectionForPvpSpawn( ply )
...
-    removeDecayTimer( ply )
-    removeDelayedRemoveTimer( ply )
+    instantRemoveSpawnProtection( ply, "You've equipped a weapon and lost spawn protection." )
+end
+
+-- Remove spawn protection on using objects
+local function spawnProtectionUseCheck( ply )
+    if not playerIsInPvp( ply ) then return end
+    if not playerHasSpawnProtection( ply ) then return end
+    instantRemoveSpawnProtection( ply, "You've used an entity and lost spawn protection." )
```
Comment (brandonsturgeon) [inline lua/autorun/server/sv_spawn_protection.lua:208]:
```suggestion
-- Remove spawn protection on using objects
local function spawnProtectionUseCheck( ply, ent )
    if not playerIsInPvp( ply ) then return end
    if not playerHasSpawnProtection( ply ) then return end
    if not IsValid( ent ) then return end
    if not ent.Use then return end

    instantRemoveSpawnProtection( ply, "You've used an entity and lost spawn protection." )
```

Maybe we should make sure the ent can be used and has a Use function?

File: lua/autorun/server/sv_spawn_protection.lua [hunk 2]
```diff
@@ -188,81 +176,80 @@ local function setSpawnProtectionForPvpSpawn( ply )
...
 
 -- Called on player keyDown events to check if a movement key was pressed
 -- and remove spawn protection if so
 local function spawnProtectionMoveCheck( ply, keyCode )
     if playerIsDisablingSpawnProtection( ply ) then return end
     if not playerHasSpawnProtection( ply ) then return end
-    if keyVoidsSpawnProtection[ keyCode ] then delayRemoveSpawnProtection( ply ) end
+    if keyVoidsSpawnProtection[ keyCode ] then
+        delayRemoveSpawnProtection( ply )
+    end
```
Comment (brandonsturgeon) [inline lua/autorun/server/sv_spawn_protection.lua:218]: Follows the RERO pattern in this repo
```suggestion
    if not keyVoidsSpawnProtection[ keyCode ] then return end

    delayRemoveSpawnProtection( ply )
```

**Review by brandonsturgeon — APPROVED** [review]
Just a couple of comments

File: lua/autorun/server/sv_spawn_protection.lua (hunk 1 above)
Comment (wrefgtzweve) [reply lua/autorun/server/sv_spawn_protection.lua:208]: I just used this as you can still pick stuff up with USE or use doors and such. Felt like those should disable it too.

File: lua/autorun/server/sv_spawn_protection.lua (hunk 2 above)
Comment (wrefgtzweve) [reply lua/autorun/server/sv_spawn_protection.lua:218]: Yeah just style changed it, should've done that too 👍

File: lua/autorun/server/sv_spawn_protection.lua (hunk 1 above)
Comment (brandonsturgeon) [reply lua/autorun/server/sv_spawn_protection.lua:208]: Sure, we'll see how it works

**Comment by brandonsturgeon** [comment]
FYI the network bools are used to let clients know who to render transparent. If you want to get rid of them, you'll need to network the status changes over to the client and update cfc_pvp

**Comment by wrefgtzweve** [comment]
> FYI the network bools are used to let clients know who to render transparent. If you want to get rid of them, you'll need to network the status changes over to the client and update cfc_pvp

https://github.com/CFC-Servers/cfc_spawn_protection/blob/9090389ac5fad9468fa11fd5060feaf06c43fa38/lua/autorun/server/sv_spawn_protection.lua#L51-L56

that's done on server side?

**Comment by brandonsturgeon** [comment]
Ah, right. That was for the outlines.

I suspect we'll end up putting them back as nwbools because we'll want some way to display their status more clearly. It's fine for now.

**Review by brandonsturgeon — APPROVED** [review]

**Pull request merged**